tox.ini

```diff
@@ -0,0 +1,70 @@
+[tox]
+envlist =
+    lint,
+    test
+
+[testenv]
+basepython = python3
+
+[testenv:test]
+deps =
+    pytest
+    pytest-girder
+commands =
+    pytest {posargs}
+
+[testenv:lint]
+skipsdist = true
+skip_install = true
+deps =
+    flake8
+    flake8-bugbear
+    flake8-docstrings
+    flake8-quotes
+    pep8-naming
+commands =
+    flake8 {posargs}
+
+[testenv:release]
+passenv =
+    TWINE_USERNAME
+    TWINE_PASSWORD
+deps =
+    twine
+commands =
+    twine check {distdir}/*
+    twine upload --skip-existing {distdir}/*
+
+[flake8]
+max-line-length = 100
+show-source = True
+format = pylint
+exclude =
+    node_modules,
+    .eggs,
+    .git,
+    __pycache__,
+    .tox
+ignore =
```

Comment:
If you want to use black, most of this can go away.  In another project we use
```
ignore =
    # closing bracket does not match indentation of opening bracket’s line
    E123
    # whitespace before ':'
    E203,
    # line break before binary operator
    W503,
    # Missing docstring in *
    D10,
```

We can add a section to auto-format:
```
[testenv:format]
skipsdist = true
skip_install = true
deps =
    black
    isort
commands =
    isort {posargs:.}
    black {posargs:.}
```

Reply:
See 54c8130b01c5f9de4e88318ea3a2e195766f31de and 19118408453d7a5ccbceffdfd146f362c8c32418.